xarray/core/common.py

```diff
@@ -212,7 +212,7 @@ def __dir__(self):
         methods.
         """
         extra_attrs = [item for sublist in self._attr_sources
-                       for item in sublist]
+                       for item in sublist if isinstance(item, str)]
```

Comment:
Great. Not putting `bytes` there as I get `E       TypeError: unorderable types: bytes() < str()`. Let me know if that's not OK though